File: src/main/java/me/arasple/mc/trmenu/utils/FileWatcher.java
package me.arasple.mc.trmenu.utils;

import io.izzel.taboolib.module.config.TConfigWatcher;

/**
 * @author Arasple
 * @date 2020/2/21 10:15
 */
public class FileWatcher {

    private static TConfigWatcher watcher;

    public static TConfigWatcher getWatcher() {
        if (watcher == null) {
            watcher = new TConfigWatcher();
        }
        return watcher;
    }

}

File: src/main/kotlin/me/arasple/mc/trmenu/utils/FileWatcher.kt
package me.arasple.mc.trmenu.utils

import io.izzel.taboolib.module.config.TConfigWatcher
import java.io.File


/**
 * @author Arasple
 * @date 2020/7/28 11:43
 */
object FileWatcher {

    fun isListening(file: File) = watcher.hasListener(file)

    fun listener(file: File, runnable: Runnable) = watcher.addSimpleListener(file, runnable)

    val watcher = TConfigWatcher()

}